### PR TITLE
Change states to unknowns

### PR DESCRIPTION
**Pull request by TorkelE**
Changes mentions of "states" to "unknowns".
Changes `get_states` (and similar functions) to `get_unknowns` (and similar functions.

I have tried to go through the full repo, and ensure that states such as in "steady state", or "the state of the system" are fine. Realistically, once all of this is over, I will probably have to read through the docs from start to finish to check that it reads consistently and nice. However, this should mostly be fine.

We cannot check whether this passes CI until MTK v9 is released. Also, CI will fail until we have figured https://github.com/SciML/ModelingToolkit.jl/issues/2459 out.

We may/may not want to merge this into https://github.com/SciML/Catalyst.jl/pull/757, but figured it would be easier to keep things somewhat separated for now.

### CODE REVIEW

**Comment by TorkelE** [comment]
I have fixed the `all_unitless` deprecation issue here as well (turning this to the PR that allows you to actually create `ReactionSystem`s using the new MTK version).

Now, we do `all(u == 1.0 for u in ModelingToolkit.get_unit([unknowns; ps; iv]))` to check whether all of `[unknowns; ps; iv]` are unitless.

**Comment by TorkelE** [comment]
I think this is ready now, tried to update the MTK thing. Tests fail, but at least Catalyst compiles. Once: https://github.com/SciML/Catalyst.jl/pull/749 we should be able to look at this now.

I try to change "state" to "unknown" where it make sense. At some point when we are done with all of this, we'll have to go through from start to finish and ensure that we are consistent with what is a state, species, unknown, variable, and whatever.

**Comment by ChrisRackauckas** [comment]
Tests fail though, it needs the `complete` calls.

File: docs/src/catalyst_applications/simulation_structure_interfacing.md [hunk 1]
```diff
@@ -89,15 +89,15 @@ Finally, we consider solution objects. First, we simulate our problem:
 ```@example ex1
 sol = solve(oprob)
 ```
-For solutions, when we access a state, we get its whole simulation vector:
+For solutions, when we access a unknown, we get its whole simulation vector:
```
[the remark this answers is not in the document]
Comment (TorkelE) [reply docs/src/catalyst_applications/simulation_structure_interfacing.md:92]: yes

File: src/networkapi.jl [hunk 2]
```diff
@@ -1434,30 +1434,30 @@ integer id of the species within the system.
...
-        sort!(get_states(rn); by = !isspecies)
+    permute!(get_unknowns(rn), neworder)
+    if !issorted(get_unknowns(rn); by = !isspecies)
+        @warn "New ordering has resulted in a non-species unknown preceding a species unknown. This is not allowed so unknowns have been resorted to ensure species precede non-species."
+        sort!(get_unknowns(rn); by = !isspecies)
     end
-    get_species(rn) .= Iterators.filter(isspecies, get_states(rn))
+    get_species(rn) .= Iterators.filter(isspecies, get_unknowns(rn))
     nothing
 end
```
[the remark this answers is not in the document]
Comment (TorkelE) [reply src/networkapi.jl:1463]: sounds good, will do

**Comment by TorkelE** [comment]
> Tests fail though, it needs the complete calls.

We plan to handle this in a separate PR (on way), this one is needed to actually compile Catalyst though, hence we are sorting it out first.

**Comment by isaacsas** [comment]
Then just merge when this is done, irrespective of the tests failing.

File: src/networkapi.jl (hunk 2 above)
[the remark this answers is not in the document]
Comment (TorkelE) [reply src/networkapi.jl]: done

**Pull request merged**